### PR TITLE
[RFC] tests: Use pending() instead of silently skipping test.

### PR DESCRIPTION
**Pull request by fwalch**

### CODE REVIEW

**Comment by justinmk** [comment]
Nice, thanks! This has been bothering me for awhile :)

File: test/functional/shell/viml_system_spec.lua [hunk 1]
```diff
@@ -184,14 +184,16 @@ describe('system()', function()
...
-        eq('', eval([[system('xclip -i -selection clipboard', 'clip-data')]]))
-        eq('clip-data', eval([[system('xclip -o -selection clipboard')]]))
-      end)
+      if xclip then
+        it('will exit properly after passing input', function()
+          eq('', eval([[system('xclip -i -selection clipboard', 'clip-data')]]))
+          eq('clip-data', eval([[system('xclip -o -selection clipboard')]]))
+        end)
+      else
+        pending('skipped (missing xclip)')
```
Comment (justinmk) [inline test/functional/shell/viml_system_spec.lua]: Although I normally favor positive logic ("if true ... else"), for validation I think it's useful to have the "short block" near the top:

```lua
if not xclip then
  pending(...)
else
  -- test logic ...
end
```

What do you think?

Comment (fwalch) [inline test/functional/shell/viml_system_spec.lua]: I think that makes sense. I'll update the PR.

File: test/functional/shell/viml_system_spec.lua [hunk 2]
```diff
@@ -184,14 +184,16 @@ describe('system()', function()
...
-      end)
+      if not xclip then
+        pending('skipped (missing xclip)')
+      else
+        it('will exit properly after passing input', function()
+          eq('', eval([[system('xclip -i -selection clipboard', 'clip-data')]]))
+          eq('clip-data', eval([[system('xclip -o -selection clipboard')]]))
+        end)
+      end
     end)
```
Comment (Pyrohh) [inline test/functional/shell/viml_system_spec.lua]: Isn't the indentation incorrect now?

Comment (Pyrohh) [inline test/functional/shell/viml_system_spec.lua]: disregard

File: test/functional/shell/viml_system_spec.lua [hunk 3]
```diff
@@ -184,14 +184,16 @@ describe('system()', function()
...
-  end
```
Comment (Pyrohh) [inline test/functional/shell/viml_system_spec.lua]: The indentation is off for this paragraph.
Assuming you saw my deleted comments, then sorry for the confusion

**Comment by Pyrohh** [comment]
Besides the indentation, LGTM.

**Comment by fwalch** [comment]
Fixed the indentation, so I'll merge this now.

**Pull request closed without merging**